lib/datetime_parser.rb

```diff
@@ -49,10 +49,7 @@ def check_time_is_valid
       return
     end
 
-    if time[:hour] > 23
-      issues << Requirements::Issue.new(field_name, :invalid)
-      return
-    end
+    issues << Requirements::Issue.new(field_name, :invalid) if time[:hour] > 23
```

Comment:
I wonder if just removing the `return` statement from this clause would have sufficed?  Though really the `ifs` should be one `if/elsif/else` block.

Reply:
Good point, I've switch to if / elsif / elsif